pubpub/pubpub
# add Collections to Pages blocks in Page editor

Resolves #1053 

Here we update the "Pages" block type in the Page editor to include both Pages and Collections.

![image](https://user-images.githubusercontent.com/2208769/95221531-85d1c100-07c5-11eb-9628-446586e426b4.png)

For now, we retain support for two block types handled by this element: `pages` (with current functionality) and `collections-pages` (supporting both Collections and Pages). The Page editor will continue to create `pages` blocks, but if you're logged in as the PubPub superadmin, you'll also see an option to create a `collections-pages` block as seen here:

![Screen Shot 2020-10-06 at 11 19 18 AM](https://user-images.githubusercontent.com/2208769/95221918-f1b42980-07c5-11eb-8043-09cb02d44e65.png)

 In a future PR I'll write a migration that turns all `pages` elements to `collections-pages` and remove codepaths supporting `pages`. This is tracked as a bullet point in #1069.

_Test plan:_
- Visit [a Page](http://localhost:9876/collections-pages-blocks-test) with `pages` and `collections-pages` blocks and verify that they render correctly. The Collection links won't work yet, which is fine.
- Visit [the corresponding Dashboard page for this Page](http://localhost:9876/dash/pages/collections-pages-blocks-test) and verify that you can add, remove, and alter both `pages` and `collections-pages` blocks. Ensure that private Collections are not leaked to logged-out users.
